### PR TITLE
Architecture check and logger improvement

### PR DESCRIPTION
**Pull request by lastranikos**
|Related issue|
|---|
| #867|

## Description

  - Issue in i386:
    - The script does not find packages for the manager in architecture: i386
    - The vagrant box, accidentally created it under 32bit (ubuntu / xenial32)

## Logs example
Script install: --ignore-health-check 
```
bash /tmp/unattended_scripts/unattended_installation.sh --AllInOne --ignore-health-check --local
12/06/2021 17:07:07 INFO: Wazuh unattended_installation: Starting process
12/06/2021 17:07:07 INFO: Configuration file found. Creating certificates...
12/06/2021 17:07:08 INFO: Creating the Elasticsearch certificates...
12/06/2021 17:07:08 INFO: Creating Wazuh server certificates...
12/06/2021 17:07:09 INFO: Creating Kibana certificate...
12/06/2021 17:07:09 INFO: Certificates creation finished. They can be found in /tmp/unattended_scripts/certs.
12/06/2021 17:07:09 WARNING: Health-check ignored.
12/06/2021 17:07:09 INFO: Prerequisites: Installing all necessary utilities for the installation...
12/06/2021 17:07:40 INFO: Prerequisites: Done
12/06/2021 17:07:40 INFO: Adding the Wazuh repository...
12/06/2021 17:07:44 INFO: Done
12/06/2021 17:07:44 INFO: Installing the Wazuh manager...
12/06/2021 17:07:44 ERROR: Wazuh installation failed
12/06/2021 17:07:44 WARNING: Cleaning the installation
12/06/2021 17:07:44 WARNING: Installation cleaned. Check the /var/log/wazuh-unattended-installation.log file to learn more about the issue.
```

**Debug: architecture**
```
uname -m
i686
```

**Debug: apt-cache**
```
apt-cache policy wazuh-*
wazuh-api:
  Installed: (none)
  Candidate: (none)
  Version table:
wazuh-agent:
  Installed: (none)
  Candidate: 4.2.5-1
  Version table:
     4.2.5-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.2.4-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.2.3-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.2.2-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.2.1-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.2.0-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.1.5-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.1.4-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.1.3-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.1.2-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.1.1-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.1.0-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.0.4-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.0.3-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.0.2-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.0.1-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
     4.0.0-1 500
        500 https://packages.wazuh.com/4.x/apt stable/main i386 Packages
wazuh-manager:
  Installed: (none)
  Candidate: (none)
  Version table:
```

## Tests remediation: --AllInOne --local -i

```
vagrant@poc-allinone-2:~$ sudo su
root@poc-allinone-2:/home/vagrant# bash /tmp/unattended_scripts/unattended_installation.sh --AllInOne --local -i
12/09/2021 12:48:56 ERROR: Uncompatible system. This script must be run on a 64-bit system.
```

## Tests remediation: --AllInOne --local

```
vagrant@poc-allinone-2:~$ sudo su
root@poc-allinone-2:/home/vagrant# bash /tmp/unattended_scripts/unattended_installation.sh --AllInOne --local
12/09/2021 12:49:02 ERROR: Uncompatible system. This script must be run on a 64-bit system.
```

### CODE REVIEW

File: .github/actions/Tests/Unattended/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+bash unattended_scripts/open-distro/unattended-installation/unattended-installation.sh
+cat /var/log/wazuh-unattended-installation.log
+sleep 60
+cd ~
+/usr/testing/bin/pytest --tb=long /test_unattended.py -v
```
Comment (rauldpm) [inline .github/actions/Tests/Unattended/entrypoint.sh:6]: Why is this added?

File: .github/actions/Tests/Unattended/pytest.ini [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+filterwarnings =
+    ignore:Unverified HTTPS request is being made.*
```
Comment (rauldpm) [inline .github/actions/Tests/Unattended/pytest.ini:3]: Same

File: .github/actions/Tests/Unattended/test_unattended.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    assert get_wazuh_api_status() == "Wazuh API REST"
+
+def test_check_log_errors():
+    found_error = False
+    with open('/var/ossec/logs/ossec.log', 'r') as f:
+        for line in f.readlines():
+            if 'ERROR' in line:
+                found_error = True
+                break
+    assert found_error == False, line
```
Comment (rauldpm) [inline .github/actions/Tests/Unattended/test_unattended.py:149]: Same

File: unattended_scripts/unattended_installation.sh [hunk 4]
```diff
@@ -51,10 +51,16 @@ importFunction() {
 }
 
 main() {
+
+    importFunction "common.sh"
+    importFunction "wazuh-cert-tool.sh"
+
```
Comment (rauldpm) [inline unattended_scripts/unattended_installation.sh:57]: Why

File: unattended_scripts/unattended_installation.sh [hunk 5]
```diff
@@ -51,10 +51,16 @@ importFunction() {
...
     if [ "$EUID" -ne 0 ]; then
         echo "Error: This script must be run as root."
         exit 1;
-    fi   
+    fi
+
+    checkArch
```
Comment (rauldpm) [inline unattended_scripts/unattended_installation.sh:63]: Call this after import the common.sh file

**Review by rauldpm — CHANGES_REQUESTED** [review]
Remove unnecessary files and call the method after the import

File: .github/actions/Tests/Unattended/entrypoint.sh (hunk 1 above)
Comment (lastranikos) [reply .github/actions/Tests/Unattended/entrypoint.sh]: This was a bad commit, I already deleted this folder.

File: .github/actions/Tests/Unattended/pytest.ini (hunk 2 above)
Comment (lastranikos) [reply .github/actions/Tests/Unattended/pytest.ini]: This was a bad commit, I already deleted this folder.

File: .github/actions/Tests/Unattended/test_unattended.py (hunk 3 above)
Comment (lastranikos) [reply .github/actions/Tests/Unattended/test_unattended.py]: This was a bad commit, I already deleted this folder.

File: unattended_scripts/unattended_installation.sh (hunk 4 above)
Comment (lastranikos) [reply unattended_scripts/unattended_installation.sh]: This was fixed in the new commit.

File: unattended_scripts/unattended_installation.sh (hunk 5 above)
Comment (lastranikos) [reply unattended_scripts/unattended_installation.sh]: This was fixed in the new commit.

**Review by rauldpm — APPROVED** [review]
LGTM

**Review by alberpilot — APPROVED** [review]
LGTM

**Pull request merged**